docs/further_useful_features.rst

```diff
@@ -0,0 +1,438 @@
+strings.
+
+In general, secondary indices behave like partitions in terms of when and how they can
+and cannot be created. However, when using ``partition_on`` the order of the columns
+provided is important, whereas it is ignored for ``secondary_indices``.
+
+
+Updating an existing dataset
+============================
+
+It's possible to update datasets by adding new physical partitions to them, ``kartothek``
+provides update functions that generally have the prefix `update_dataset` in their names.
+For example, :func:`~kartothek.io.eager.update_dataset_from_dataframes` is the update
+function for the ``eager`` backend.
+
+To see updating in action, lets set up some data and a storage location first and store
+the data there with ``kartothek``:
+
+.. ipython:: python
+
+    dm = store_dataframes_as_dataset(store_factory, "a_unique_dataset_identifier", df)
+    dm.partitions
+
+
+Now, we create ``another_df`` with the same schema as our intial dataframe
+``df`` and update it using the ``eager`` backend by calling :func:`~kartothek.io.eager.update_dataset_from_dataframes`:
+
+.. ipython:: python
+
+    from kartothek.io.eager import update_dataset_from_dataframes
+
+    another_df = pd.DataFrame(
+        {
+            "A": 5.,
+            "B": pd.Timestamp("20110102"),
+            "C": pd.Series(2, index=list(range(4)), dtype="float32"),
+            "D": np.array([6] * 4, dtype="int32"),
+            "E": pd.Categorical(["prod", "dev", "prod", "dev"]),
+            "F": "bar",
+        }
+    )
+
+    dm = update_dataset_from_dataframes(
+        [another_df],
+        store=store_factory,
+        dataset_uuid="a_unique_dataset_identifier"
+    )
+    dm.partitions
+
+
+Looking at ``dm.partitions``, we can see that another partition has
+been added.
+
+If we read the data again, we can see that the ``another_df`` has been appended to the
+previous contents.
+
+.. ipython:: python
+
+    from kartothek.io.eager import read_table
+
+    updated_df = read_table("a_unique_dataset_identifier", store_factory, table="table")
+    updated_df
+
+
+The way dataset updates work is that new partitions are added to a dataset
+as long as they have the same tables as the existing partitions. A `different`
+table **cannot** be introduced into an existing dataset with an update.
+
+To illustrate this point better, lets first create a dataset with two tables:
+
+.. ipython:: python
+
+    df2 = pd.DataFrame(
+        {
+            "G": "foo",
+            "H": pd.Categorical(["test", "train", "test", "train"]),
+            "I": np.array([9] * 4, dtype="int32"),
+            "J": pd.Series(3, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190604"),
+            "L": 2.,
+        }
+    )
+    df2
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "another_unique_dataset_identifier",
+        dfs = [
+            {
+                "data":
```

Comment:
I'd like to see a section here mentioning that the usage of the explicit label is discouraged here since for the update to properly work using copy-on-write principles we require the filenames to be unique. This is why kartothek uses UUIDs for the labels by default

Reply:
Also add a comment about future deprecation of explicit labels?

Reply:
Wouldn't mentioning that it is possible to use explicit labels go against the point of not talking about features we do not want users to use?

Reply:
Atm https://github.com/JDASoftwareGroup/kartothek/pull/62/commits/2d450bfaabcd67d3cdcecc72606f486b6cb36662 adds mention of why UUID's are used as labels, does not talk about explicit labels